jerry-ext/module/module.c

```diff
@@ -0,0 +1,215 @@
+{
+  jerry_release_value (*(jerry_value_t *) user_data_p);
+} /* jerryx_module_manager_deinit */
+
+/**
+ * Declare the context data manager for modules.
+ */
+static const jerry_context_data_manager_t jerryx_module_manager =
+{
+  .init_cb = jerryx_module_manager_init,
+  .deinit_cb = jerryx_module_manager_deinit,
+  .bytes_needed = sizeof (jerry_value_t)
+};
+
+/**
+ * Declare the linker section where module definitions are stored.
+ */
+JERRYX_SECTION_DECLARE (jxm_modules, jerryx_module_t)
+
+/**
+ * Attempt to retrieve a module by name from a cache, and return zero if not found.
+ */
+static bool
+jerryx_module_check_cache (jerry_value_t cache, /**< cache from which to attempt to retrieve the module by name */
+                          jerry_value_t module_name, /**< JerryScript string value holding the module name */
+                          jerry_value_t *result) /**< Resulting value */
+{
+  bool ret = false;
+
+  /* Check if the cache has the module. */
+  jerry_value_t js_has_property = jerry_has_property (cache, module_name);
+
+  /* If we succeed in getting an answer, we examine the answer. */
+  if (!jerry_value_has_error_flag (js_has_property))
+  {
+    bool has_property = jerry_get_boolean_value (js_has_property);
+
+    /* If the module is indeed in the cache, we return it. */
+    if (has_property)
+    {
+      (*result) = jerry_get_property (cache, module_name);
+      ret = true;
+    }
+  }
+
+  jerry_release_value (js_has_property);
+
+  return ret;
+} /* jerryx_module_check_cache */
+
+/**
+ * Attempt to cache a loaded module.
+ *
+ * @return the module on success, otherwise the error encountered when attempting to cache. In the latter case, the
+ * @p module is released.
+ */
+static jerry_value_t
+jerryx_module_add_to_cache (jerry_value_t cache, /**< cache to which to add the module */
+                            jerry_value_t module_name, /**< key at which to cache the module */
+                            jerry_value_t module) /**< the module to cache */
+{
+  jerry_value_t ret = jerry_set_property (cache, module_name, module);
+
+  if (jerry_value_has_error_flag (ret))
+  {
+    jerry_release_value (module);
+  }
+  else
+  {
+    jerry_release_value (ret);
+    ret = module;
+  }
+
+  return ret;
+} /* jerryx_module_add_to_cache */
+
+/**
+ * Declare and define the default module resolver - one which examines what modules are defined in the above linker
+ * section and loads one that matches the requested name, caching the result for subsequent requests using the context
+ * data mechanism.
+ */
+JERRYX_MODULE_RESOLVER (native_resolver)
+{
+  int index;
+  const jerryx_module_t *module_p = NULL;
+
+  /* Look for the module by its name in the list of module definitions. */
+  for (index = 0, module_p = &__start_jxm_modules[0];
+  &__start_jxm_modules[index] < __stop_jxm_modules;
+  index++, module_p = &__start_jxm_modules[index])
```

Comment:
Skimming over the coding standards doc I don't see any section that applies to long for-statement headers. How should I indent this? Keep in mind that `check-vera.sh` does not now complain about indentation.